app/scripts/metamask-controller.js

```diff
@@ -360,6 +361,10 @@ export default class MetamaskController extends EventEmitter {
       ///: END:ONLY_INCLUDE_IN
     });
 
+    this.preferencesController.store.subscribe(async ({ currentLocale }) => {
```

Comment:
Nit: If we export `currentLocale` from `translate.ts`, we could compare with the `updatedLocale` here and then `updateCurrentLocale` in order to prevent unnecessary updates.

Reply:
Ah true, missed that.